tests/distributions/test_hmm.py

```diff
@@ -55,6 +55,15 @@ def check_expand(old_dist, old_data):
     assert new_dist.log_prob(new_data).shape == new_batch_shape
 
 
+def check_sample_shape(d):
```

Comment:
shouldn't we have a more comprehensive test? e.g. create a `duration=1` hmm with posterior mass that concentrates on `x=0` and test accordingly?

Reply:
Thanks for keeping me honest! I've added a test of distribution with a simple example as you suggest.